main/http.c

```diff
@@ -106,9 +114,15 @@ http_send(const char *const p_msg)
         return false;
     }
 
-    esp_http_client_set_post_field(http_handle, p_msg, strlen(p_msg));
+    esp_http_client_set_post_field(http_handle, p_msg, (int)strlen(p_msg));
     esp_http_client_set_header(http_handle, "Content-Type", "application/json");
 
+    const hmac_sha256_str_t hmac_sha256 = hmac_sha256_calc(p_msg);
+    if ('\0' != hmac_sha256.buf[0])
```

Comment:
Does something guarantee that first byte of sha256 buffer is never 0 on successful calculation? It's my understanding that all values 0 ... 255 are equally likely, which would cause this to fail 1/255 times

Reply:
it's a string, so if it's not empty it means that it contains hex-encoded HMAC_SHA256.
In any case, I've refactored this part to avoid confusion.